### ISSUE TITLE
[FEATURE] Option for no line wraps in LDIF export method

### ISSUE THREAD

**Issue by jaybeers** [cannatag/ldap3#852]
This is in reference to the method `conn.response_to_ldif()` documented at https://ldap3.readthedocs.io/en/latest/connection.html#connection; to save a click:

> response_to_ldif: a method you can call to convert the response of a search to a LDIF format (ldif-content). It has the following parameters:
>
>         search_result: the result of the search to be converted (defaults to None). If None get the last response received from the Server
>         all_base64: converts all the value to base64 (defaults to False)

What I would like is an additional parameter for this method, to suppress all line wrapping.  For example, instead of this:

```
version: 1
dn: unique_id=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,ou=users,o=objects,dc=company,dc
 =com
objectClass: Metadata
objectClass: Person
uid: testuser
```
...I'd like this (note that the `dn` line isn't wrapped):
```
version: 1
dn: unique_id=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,ou=users,o=objects,dc=company,dc=com
objectClass: Metadata
objectClass: Person
uid: testuser
```

I'm not referring to the natural line breaks, empty lines, etc. that occur as part of the LDIF file format; just the ones that get inserted to break up long lines.  Basically, what you get with `ldapsearch -o ldif-wrap=no`.

**Comment by cannatag** [comment]
Hi, the 78 chars length of LDIF statements is fixed and defined in RFC 2849. Anyway in the ldap3/protocol/rfc2849.py file there is the LDIF_LINE_LENGTH = 78 parameter that you can change.

I will add a configuration parameter for this in the next release.

Bye,
Giovanni

**Comment by cannatag** [comment]
I've added the LDIF_LINE_LENGTH configuration parameter. Now you can simply:

```
from ldap3.utils.config import set_config_parameter
set_config_parameter('LDIF_LINE_LENGTH', 200)
```
and the line will be wrapped at 200 chars. Or whatever number you want.

Can you check the code in dev?

thanks,
Giovanni

**Comment by cannatag** [comment]
Fixed in ldap3 v2.8.

**Comment by Toxsch** [comment]
I encountered this problem as well and believe there is a minor issue with the implementation.

In rfc2849.py, the conf_ldif_line_length variable is defined as a global and initialized when the module is loaded. 
```python
# rfc2849.py:35
# LDIF converter RFC 2849 compliant

conf_ldif_line_length = get_config_parameter('LDIF_LINE_LENGTH')
```

As a result, even if the LDIF_LINE_LENGTH configuration changes, the conf_ldif_line_length variable still appears to default to 78.
```python
# rfc2849.py:219
def operation_to_ldif(operation_type, entries, all_base64=False, sort_order=None):
    ....
    # check max line length and split as per note 2 of RFC 2849
    for line in lines:
        if line:
            ldif_record.append(line[0:conf_ldif_line_length])
            ldif_record.extend([' ' + line[i: i + conf_ldif_line_length - 1] for i in range(conf_ldif_line_length, len(line), conf_ldif_line_length - 1)] if len(line) > conf_ldif_line_length else [])
        else:
            ldif_record.append('')

    return ldif_record
```
